cmd/thanos/query.go

```diff
@@ -91,6 +91,9 @@ func registerQuery(m map[string]setupFunc, app *kingpin.Application, name string
 	enablePartialResponse := cmd.Flag("query.partial-response", "Enable partial response for queries if no partial_response param is specified.").
 		Default("true").Bool()
 
+	storeReceiveTimeout := modelDuration(cmd.Flag("store.receive-timeout", "Maximum time to wait for any data from store. If Store doesn't send any data any storeReceiveTimeout the Store will be igored and partial data will be returned.").
+		Default("200ms"))
```

Comment:
wonder if it should not be disabled by default to not surprise people with success on query API and actually partial data (:

Reply:
Yeah I agree, I think having to explicitly opt-in for this would be best so people are definitely aware of the possibility of partial data